### PR TITLE
Migrate match-quote module to TS

### PR DESCRIPTION
**Pull request by acelaya**
This migrates match-quote to TS

### CODE REVIEW

File: src/annotator/anchoring/match-quote.ts [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
- * @prop {number} score -
- *   Score for the match between 0 and 1.0, where 1.0 indicates a perfect match
- *   for the quote and context.
- */
+type Match = {
+  /** Start offset of match in text */
+  start: number;
+  /** End offset of match in text */
+  end: number;
+  /**
```
Comment (lyzadanger) [inline src/annotator/anchoring/match-quote.ts:9]: World's most pedantic PR comment, and one you can take or leave: I tend to leave an extra line break above multi-line property comments. So:

```
type Whatever = {
  /** Short comment about this property **/
  foo: string;
  /** Short comment about this property **/
  bar: string;

  /**
   * A multi line...
   * ...
   * comment about this property
   */
  baz: number;
}
```;

File: src/annotator/anchoring/match-quote.ts [hunk 2]
```diff
@@ -66,22 +59,29 @@ function textMatchScore(text, str) {
...
  * @param {string} text - Document text to search
  * @param {string} quote - String to find within `text`
- * @param {object} context -
- *   Context in which the quote originally appeared. This is used to choose the
- *   best match.
- *   @param {string} [context.prefix] - Expected text before the quote
- *   @param {string} [context.suffix] - Expected text after the quote
- *   @param {number} [context.hint] - Expected offset of match within text
- * @return {Match|null}
+ * @param context - Context in which the quote originally appeared. This is
```
Comment (lyzadanger) [inline src/annotator/anchoring/match-quote.ts:76]: You can remove the redundant type hint for the preceding `@param`s.

**Review by lyzadanger — APPROVED** [review]
👍🏻

File: src/annotator/anchoring/match-quote.ts (hunk 2 above)
Comment (acelaya) [reply src/annotator/anchoring/match-quote.ts:76]: Right!

File: src/annotator/anchoring/match-quote.ts (hunk 1 above)
Comment (lyzadanger) [reply src/annotator/anchoring/match-quote.ts:10]: Oh, I wouldn't expect any of our code to actually be _consistent_ in this respect, hah!

**Pull request merged**